spec/chewy/strategy/lazy_sidekiq_spec.rb

```diff
@@ -0,0 +1,230 @@
+        expect(city).to receive(:run_chewy_callbacks).and_call_original
+        expect(other_city).to receive(:run_chewy_callbacks).and_call_original
+
+        ::Sidekiq::Testing.inline! do
+          Chewy::Strategy::LazySidekiq::IndicesUpdateWorker.new.perform({'City' => [city.id, other_city.id]})
+        end
+      end
+
+      context 'when Chewy.disable_refresh_async is true' do
+        before do
+          allow(Chewy).to receive(:disable_refresh_async).and_return(true)
+        end
+
+        it 'calls Index#import! with refresh false' do
+          expect(CitiesIndex).to receive(:import!).with([city.id, other_city.id], refresh: false)
+          expect(::Sidekiq::Client).to receive(:push)
+            .with(hash_including('class' => Chewy::Strategy::Sidekiq::Worker, 'queue' => 'low'))
+            .and_call_original
+
+          allow(City).to receive(:where).with(id: [city.id, other_city.id]).and_return([city, other_city])
+          expect(city).to receive(:run_chewy_callbacks).and_call_original
+          expect(other_city).to receive(:run_chewy_callbacks).and_call_original
+
+          ::Sidekiq::Testing.inline! do
+            Chewy::Strategy::LazySidekiq::IndicesUpdateWorker.new.perform({'City' => [city.id, other_city.id]})
+          end
+        end
+      end
+    end
+
+    context 'integration' do
+      around { |example| ::Sidekiq::Testing.inline! { example.run } }
+
+      let(:update_condition) { true }
+
+      before do
+        city_model
+        country_model
+
+        City.belongs_to :country
+        Country.has_many :cities
+
+        stub_index(:cities) do
+          index_scope City
+        end
+
+        stub_index(:countries) do
+          index_scope Country
+        end
+      end
+
+      context 'state dependent' do
+        let(:city_model) do
+          stub_model(:city) do
+            update_index(-> { 'cities' }, :self)
+            update_index('countries') { changes['country_id'] || previous_changes['country_id'] || country }
+          end
+        end
+
+        let(:country_model) do
+          stub_model(:country) do
+            update_index('cities', if: -> { state_dependent_update_condition }) { cities }
+            update_index(-> { 'countries' }, :self)
+            attr_accessor :state_dependent_update_condition
+          end
+        end
+
+        context 'city updates' do
+          let!(:country1) { Country.create!(id: 1) }
+          let!(:country2) { Country.create!(id: 2) }
+          let!(:city) { City.create!(id: 1, country: country1) }
+
+          it 'does not update index of removed entity because model state on the moment of save cannot be fetched' do
+            expect { city.update!(country: nil) }.not_to update_index('countries', strategy: :lazy_sidekiq)
+          end
+          it 'does not update index of removed entity because model state on the moment of save cannot be fetched' do
```

Comment:
It does match, but probably the description is just confusing?
We had `city` with `country1` and we update it with `country2`, it will reindex `country2` only.
Similar test from any other strategy that keeps the state would look like:
```ruby
expect { city.update!(country: country2) }.to update_index('countries', strategy: :lazy_sidekiq).and_reindex(country1, country2).only
```

Reply:
I see, thanks. Maybe confusing a bit. I thought test name echoes `state_dependent_update_condition` somehow.

Reply:
Yeah, the only test that uses `state_dependent_update_condition` is `does not update index because state of attribute cannot be fetched` and it demonstrates that if condition is state dependent index won't be updated.
I'll try to come up with more descriptive context and example names.